robot_sort/robot_sort.py

```diff
@@ -93,11 +93,39 @@ def light_is_on(self):
         return self._light == "ON"
 
     def sort(self):
-        """
-        Sort the robot's list.
-        """
-        # Fill this out
-        pass
+        while self.light_is_on() == False:  
+            self.set_light_on()
+
+            if self.can_move_right() == False:
+                while self.can_move_left() == True:
+
+                    if self.compare_item() == None:
+                        self.swap_item()
+                        self.move_left()
+
+                    if self.compare_item() == -1:
+                        self.swap_item()
+                        self.move_right()
+
+                    elif self.compare_item() == 1 or self.compare_item() == 0:
+                        self.move_right()
+
+            while self.can_move_right() == True:
+
+                if self.compare_item() == None:
+                    self.swap_item()
+                    self.move_right()
+
+                if self.compare_item() == 1:
+                    self.swap_item()
+                    self.move_left()
+                    self.set_light_off()
+
+                elif self.compare_item() == -1 or self.compare_item() == 0:
+                    self.move_left()
+                          
+        return self._list
+
```

Comment:
Good job implementing this. An alternative approach would have been a selection sort which could have been done like this:

self.swap_item()
self.set_light_on()

while self.light_is_on():
(move right and find the smallest item that has not been sorted)
____while self.move_right():
________if self.compare_item() > 0:
____________self.swap_item()
( if the robot is at the end of the list and all items are sorted, we're done)
________if self.compare_item() is None and self.can_move_right() is False:
(place the current item down in the list and turn light off to indicate the list is sorted)
____________self.swap_item()
____________self.set_light_off()
____________break
________else:
( find the spot in the list that is holding None)
( place the item in that position)
(move right and swap)
____________while self.move_left():
________________if self.compare_item() is None:
________________self.swap_item()
________________self.move_right()
________________self.swap_item()
________________break